### PR TITLE
Refactor fix

### PR DESCRIPTION
**Pull request by mcat56**
Clean up code, remove commented out code
Add uuid for api key and bcrypt for user password
Add password to user table

### CODE REVIEW

**Comment by mcat56** [comment]
@BabsLabs

File: app/services/forecast_service.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ class ForecastService {
     })
     .catch((error) => {
       console.log({error:error, location:'getGeInfo'});
-      return {error_message: error};
+      return {error_message: error.message};
```
Comment (BabsLabs) [inline app/services/forecast_service.js:23]: Nice use of sending the error.message. That makes it much easier for the user to understand the error messages.

File: db/seeds/dev/users.js [hunk 2]
```diff
@@ -8,9 +8,9 @@ exports.seed = function(knex) {
       return Promise.all([
 
         knex('users').insert([
-          {api_key: '123456789', email: 'whereswaldo123@gmail.com' },
-          {api_key: '987654321', email: 'carmensandiego456@gmail.com' },
-          {api_key: '024681357', email: 'harrietthespy789@gmail.com' }
+          {api_key: '123456789', email: 'whereswaldo123@gmail.com', password: 'seemenot' },
+          {api_key: '987654321', email: 'carmensandiego456@gmail.com', password: 'catchmeifyoucan' },
+          {api_key: '024681357', email: 'harrietthespy789@gmail.com', password: 'doubleagent' }
```
Comment (BabsLabs) [inline db/seeds/dev/users.js:13]: This was one of my favorite books growing up! The movie was alright too but the book was great!

File: routes/api/v1/favorites.js [hunk 3]
```diff
@@ -57,14 +57,14 @@ router.post('/', (req, res) => {
     .then((user) => {
       database('favorites').insert({ location: location, user_id: user.id }, 'location')
       .then((location) => {
-        res.status(200).json({ message: `${location} has been added to your favorites`})
+        res.status(201).json({ message: `${location} has been added to your favorites`})
```
Comment (BabsLabs) [inline routes/api/v1/favorites.js:60]: Again, good use of using the proper error code.

File: routes/api/v1/forecasts.js [hunk 4]
```diff
@@ -11,50 +11,27 @@ const fetch = require('node-fetch')
...
         getForecast()
-      } else {
+      } else if (!req.body.api_key ){
         res.status(401).json({error_message: 'Unauthorized request' })
+      } else if (!req.query.location){
+        res.status(400).json({error_message: 'Location required' })
       }
     })
     .catch((error) => {
       console.log({error:error, location: 'forecast get'});
```
Comment (BabsLabs) [inline routes/api/v1/forecasts.js:33]: Maybe change the `location: 'location get'` part of the error. I don't understand it on a quick glance but it probably isn't a big deal.

File: routes/api/v1/users.js [hunk 5]
```diff
@@ -12,8 +14,27 @@ router.get('/', (req, res) => {
       res.status(200).json(users);
     })
     .catch((error) => {
-      res.status(500).json({error_message: error});
+      res.status(500).json({error_message: error.message});
+    });
+});
+
+router.post('/', (req, res) => {
```
Comment (BabsLabs) [inline routes/api/v1/users.js:21]: Way to do the extension!

**Review by BabsLabs — COMMENTED** [review]
Great work. Your error handling was excellent and you used the proper status codes when appropriate. I'm also quite impressed that you went ahead and added the post user endpoint for the extension.

Overall, I think this great and am going to approve the PR.

**Review by BabsLabs — APPROVED** [review]

File: routes/api/v1/users.js (hunk 5 above)
Comment (mcat56) [reply routes/api/v1/users.js:21]: Thank you!

File: routes/api/v1/forecasts.js (hunk 4 above)
Comment (mcat56) [reply routes/api/v1/forecasts.js:33]: @BabsLabs 
It was to help see where my errors were coming from, just a label in the console essentially, but I will be taking that out!

**Pull request merged**